### ISSUE TITLE
Disabled checkbox should not be in the tab order

### ISSUE THREAD

**Issue by marcysutton** [angular/material#2087]
To better align with native checkboxes, `mdCheckbox` components using `ng-disabled` should not have `tabIndex=0`. A native checkbox will not be part of the tab order when the `disabled` attribute is used. The Material Checkbox should have similar behavior. 

@ThomasBurleson started PR #1168 for `ng-disabled` improvements for `mdButton`, but I wanted to capture the `mdCheckbox` tabindex issue. `mdSwitch` does not suffer from the same problem. 

See the "mdCheckbox" section in this Codepen: http://codepen.io/marcysutton/pen/yyxREg
Compare it to this native checkbox Codepen: http://codepen.io/marcysutton/pen/GgxboO

**Comment by ThomasBurleson** [comment]
Related to #2297
